apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -289,6 +290,31 @@ public int Timeout
             set => SetInt32WithValidation(nameof(Timeout), 0, int.MaxValue, value);
         }
 
+        /// <summary>
+        /// Specifies whether to allow the connection to check for the presence of the emulator
+        /// environment variable.
+        /// </summary>
+        /// <remarks>
+        /// This property defaults to <see cref="EmulatorDetection.None"/>, meaning that the
+        /// environment variable is ignored.
+        /// </remarks>
+        public EmulatorDetection EmulatorDetection
+        {
+            get
+            {
+                if (TryGetValue(EmulatorDetectionKeyword, out object value) &&
+                    (value is EmulatorDetection parsed || (value is string text && EmulatorDetection.TryParse(text, out parsed))))
+                {
+                    return parsed >= EmulatorDetection.None && parsed <= EmulatorDetection.EmulatorOrProduction ? parsed : EmulatorDetection.None;
+                }
+                return EmulatorDetection.None;
+            }
+            set
+            {
+                this[EmulatorDetectionKeyword] = (int) value;
```

Comment:
Setting it to the integer value will make the connection string very hard to read in string form. I would call ToString - after validating that it's a valid value using `GaxPreconditions.CheckEnumValue`.